src/headers.rs

```diff
@@ -120,16 +120,15 @@ impl TryFrom<&[u8]> for Headers {
         };
 
         for line in lines {
-            let splits = line.splitn(2, ':').map(str::trim).collect::<Vec<_>>();
-            match splits[..] {
-                [k, v] => {
-                    let entry = inner.entry(k.to_string()).or_insert_with(HashSet::default);
-                    entry.insert(v.to_string());
-                }
-                [""] => continue,
-                _ => {
-                    return parse_error("malformed header input");
-                }
+            if line.is_empty() {
+                continue;
+            }
+
+            if let Some((k, v)) = line.split_once(':') {
```

Comment:
@derekcollison Don't we want to have graceful behaviour here and trim, as in previous implementation?

Reply:
Yes..